scripts/setup-gcloud-cluster.sh

```diff
@@ -125,26 +129,118 @@ gcloud_create_or_update_managed_instance_group  \
+declare -a used_staking_addrs
+
+# Iterate through all VM instances
+# The loop should be parallelized in future to accommodate better with larger clusters
+for instance_idx in "${!instance_names_arr[@]}" ; do
+  # Firstly, retrieve the IP address of this VM instance
+  instance_name="${instance_names_arr[instance_idx]}"
+  node_ip=$(gcloud_get_ip "${instance_name}")
+
+  # All VM instances in the deployed cluster will get a special INFO tag
+  # which contains all handy information about the HOPR instance running in the VM.
+  # These currently include: node wallet address, node peer ID, associated staking account
+  # These information are constant during the lifetime of the VM and
+  # do not change during re-deployment.
+  info_tag=$(gcloud_get_node_info_tag "${instance_name}")
+
+  # Values contained in the INFO tag
+  declare wallet_addr
+  declare peer_id
+  declare staking_addr
+  if [[ -z "${info_tag}" ]]; then
+    # If the instance does not have the INFO tag yet, we need to retrieve all info
+    wallet_addr=$(get_native_address "${api_token}@${node_ip}:3001")
+    peer_id=$(get_hopr_address "${api_token}@${node_ip}:3001")
+
+    # NOTE: We leave only the first public node unstaked
+    if [[ ${instance_idx} -eq 0 && "${docker_image}" != *-nat:* ]]; then
+      staking_addr="unstaked"
+    else
+      # Staking accounts are assigned round-robin
+      staking_addr_idx=$(( (instance_idx ) % ${#staking_addresses_arr[@]} ))
+      staking_addr="${staking_addresses_arr[staking_addr_idx]}"
+    fi
+
+    # Save the info tag
+    info_tag="info:native_addr=${wallet_addr};peer_id=${peer_id};nr_staking_addr=${staking_addr}"
+    gcloud_add_instance_tags "${instance_name}" "${info_tag}"
+  else
+    # Retrieve all information from the INFO tag
+    wallet_addr=$(echo "${info_tag}" | sed -E 's/.*native_addr=(0[xX]{1}[a-f0-9A-F]{40}).*/\1/g')
+    peer_id=$(echo "${info_tag}" | sed -E 's/.*peer_id=([a-zA-Z0-9]+).*/\1/g')
+    staking_addr=$(echo "${info_tag}" | sed -E 's/.*nr_staking_addr=([a-zA-Z0-9]+).*/\1/g')
+  fi
+
+  ip_addrs+=( "${node_ip}" )
+
+  # Do not include the unstaked nodes (= skipped during registration for NR)
+  if [[ "${staking_addr}" != "unstaked" ]]; then
+    hopr_addrs+=( "${peer_id}" )
+    used_staking_addrs+=( "${staking_addr}" )
+  fi
+
+  # Fund the node as well
+  wait_until_node_is_ready "${node_ip}"
+  fund_if_empty "${wallet_addr}" "${environment}"
+
 done
 
-# To test Network registry, the cluster_size is greater or equal to 2 and staker_addresses are provided as parameters
-# TODO: call stake API so that the first staker_addresses[0] stake in the current program
-# TODO: call register API and register staker_addresses with node peer ids
+# Register all nodes in cluster
+IFS=','
+# If same order of parameters is given, the "register" task is idempotent
+yarn workspace @hoprnet/hopr-ethereum hardhat register \
+   --network hardhat \
+   --task add \
+   --native-addresses "${used_staking_addrs[*]}" \
```

Comment:
Not sure if `--native-addresses` works or only `--nativeAddresses` can be accepted (or the other way around.) Similar for `--peer-ids`

Reply:
Checked other tasks (such as `faucet`) and the version with `-` should work fine.